### PR TITLE
TEST-#4550: Use much less data in test_partition_api

### LINKED ISSUE #4550: TEST: Figure out why modin/test/test_partition_api.py with 2 pytest cores is so flaky.
The original issue is #4508. When I ran this test on an ubuntu machine I saw object store memory going up and some spilling to disk. Maybe that's related to why setting 2 pytest cores is flaky. The test takes a while with 1 core, so it would be good to re-enable 2 cores again. For now in the fix for #4508, I will move the test to 1 core.

### PR DESCRIPTION
**Pull request by mvashishtha**
## What do these changes do?

TEST-#4550: Use much less data in test_partition_api.

We don't need to test the partition API with such huge data. The large data made the test slow, which is why we originally why we had 2 cores. The 2 ray clusters started in the 2 cores probably could not handle all the data, resulting in #4508. We should just test with enough data to fill all the partitions. `test_data['int_data']` from the test utils should be good enough.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4550
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Comment by mvashishtha** [comment]
I need to wait for #4551 to be merged  so I can delete the comment about understanding the ncores bug. Converted this PR to draft.

**Review by prutskov — DISMISSED** [review]
LGTM!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -257,9 +257,6 @@ jobs:
       - run: python -m pytest modin/test/storage_formats/base/test_internals.py
       - run: python -m pytest modin/test/storage_formats/pandas/test_internals.py
       - run: python -m pytest modin/test/test_envvar_npartitions.py
-      # TODO(https://github.com/modin-project/modin/issues/4550): this test is
-      # slow (~140 sec), but using 2 cores usually causes ray worker crashes.
-      # Fix the crashes and increase cores to 2.
       - run: python -m pytest modin/test/test_partition_api.py
```
Comment (YarShev) [inline .github/workflows/ci.yml:260]: Should we return back pytest -n 2 for modin/test/test_partition_api.py?

Comment (mvashishtha) [reply .github/workflows/ci.yml:260]: The test is so fast now that `-n 2` seems to actually slow the test down. on my mac, I get 24.07 sec with `-n 2`, and 13.3 sec with the default of 1. I'm on 

- macOSMonterey 12.2.1
- 2.3 GHz 8-Core Intel Core i9
- 16 GB 2667 MHz DDR4

Comment (YarShev) [reply .github/workflows/ci.yml:260]: Ok, but your machine is different from those machines that are used in GH actions. It seems to me those have 2 CPUs each. I would suggest re-checking the need of `pytest -n 2 usage` for tests to understand if we really see the benefit. Of course, as part of a different issue.

**Review by YarShev — APPROVED** [review]
@mvashishtha, LGTM, thanks!

File: docs/release_notes/release_notes-0.16.0.rst [hunk 2]
```diff
@@ -23,6 +23,7 @@ Key Features and Updates
   *
 * Update testing suite
   * TEST-#4508: Reduce test_partition_api pytest threads to deflake it (#4551)
+  * TEST-#4550: Use much less data in test_partition_api (#4554).
```
Comment (YarShev) [inline docs/release_notes/release_notes-0.16.0.rst:26]:
```suggestion
  * TEST-#4550: Use much less data in test_partition_api (#4554)
```

**Review by YarShev — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (prutskov) [reply .github/workflows/ci.yml:260]: I agree with @YarShev that it's need to definne the need of using 2 parallel jobs with Ray/Dask clusters on 2 cores GH servers. I was also faced with issues when `-n 2` only increase execution time.

**Pull request merged**